src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/DefaultPageFactory.cs

```diff
@@ -61,6 +61,7 @@ public class DefaultPageFactory : IPageFactoryProvider
             {
                 var page = (Page)activatorFactory(context);
                 page.PageContext = context;
+                page.Path = context.ActionDescriptor.ViewEnginePath;
```

Comment:
Sorry, I was looking at the docs for the `IRazorPage.Path` and it says it's the app relative path. `ActionDescriptor.RelativePath` is probably the better value here since it's not based on `RazorPagesOptions.BasePath`. Could you update this prior to checking this in?